cirq/contrib/paulistring/clifford_optimize.py

```diff
@@ -100,7 +100,8 @@ def try_merge_clifford(cliff_op: ops.GateOperation, start_i: int) -> bool:
 
             qubit, pauli = next(iter(merge_op.pauli_string.items()))
             quarter_turns = round(merge_op.half_turns * 2)
-            quarter_turns *= (1, -1)[merge_op.pauli_string.negated]
+            assert merge_op.pauli_string.coefficient in [1, -1]
```

Comment:
Alternatively, you could keep the `negated` method and make it check the coefficient value, raise if coefficient's imaginary part is non-zero, else return its real part.

Reply:
Added a TODO.